docs/1.getting-started/7.state-management.md

```diff
@@ -57,9 +57,67 @@ const counter = useState('counter', () => Math.round(Math.random() * 1000))
 To globally invalidate cached state, see [`clearNuxtState`](/docs/api/utils/clear-nuxt-state) util.
 ::
 
-### Advanced Usage
+### Init with Async Data
 
-In this example, we use a composable that detects the user's default locale from the HTTP request headers and keeps it in a `locale` state.
+Most of the time you will want to initialize your state with async data, you can use the [`app.vue`](/docs/guide/directory-structure/app) component to do so with the [`once`](/docs/api/utils/once) util.
+
+```vue [app.vue]
+<script setup>
+const websiteConfig = useState('config')
+
+await once(async () => {
+  websiteConfig.value = await $fetch('https://my-cms.com/api/website-config')
+})
+</script>
+```
+
+::callout
+This is similar to Nuxt 2 `nuxtServerInit` action to fill the store with data before rendering the page.
+::
+
+:read-more{to="/docs/api/utils/once"}
+
+### Usage with Pinia
+
+In this example, we leverage the [Pinia module](/modules/pinia) to create a global store and use it across the app.
+
+::callout
+Make sure to install the Pinia module with `npx nuxi@latest module add pinia`
```

Comment:
Should this also include regular installation steps for those that don't use the cli for whatever reason?

Reply:
I link to the Pinia module actually so I think that should be enough no?

Reply:
I guess so. Perhaps it can say something like "or follow the module's installation stete"? Some people might mistake it for a typo in the docs if it's not made clear that it's equivalent, especially those whose native language isn't English.

Reply:
Updated